### PR TITLE
feat(mobile): add SureButton primitive and migrate login buttons

### LINKED ISSUE #2357: Add a SureButton design-system primitive (off Material)
## Summary

Part of the mobile design-system sequence (#2235) — the first Phase-1 primitive. The app draws CTAs with Material `ElevatedButton`/`OutlinedButton`/`TextButton`, which carry Material styling (ripple, accent-colored outline/text foregrounds) rather than the Sure design system. Introduce a custom `SureButton` mirroring the web `DS::Button`, off Material.

This is **independent** of the in-review foundation PRs — it uses only tokens already on `main`.

## Scope

- **`SureButton` primitive** mirroring `DS::Button` / `DS::Buttonish`: variants (`primary`, `secondary`, `destructive`, `outline`, `ghost`), sizes (`sm`/`md`/`lg` ≈ 28/36/48), optional leading widget, `fullWidth`, `loading`, and disabled. Colors resolve from the `SureColors` palette; the shape from `radiusMd`/`radiusLg` tokens; flat custom press feedback (no Material ripple).
- **Migrate the 3 main-view login buttons** as the showcase: Sign In (primary, lg, loading), Sign in with Google (outline + SVG leading), API-Key Login (ghost).
- Out of scope (follow-ups): the 2 buttons inside the API-key **modal dialog**, and the remaining button call sites across settings / transactions / chat / etc.

## Why

One consistent, off-Material button matching the web DS — neutral (not Material-accent) outline/ghost foregrounds, tokenized colors, DS shape — and the foundational primitive that dialogs, forms, empty states, and CTAs all build on.

## Acceptance

- `flutter analyze --no-fatal-infos` clean (no new issues), `flutter test` green, and `flutter build ios --simulator --debug` + `flutter build apk --debug` pass.
- Before/after login screenshots (light + dark) show the migration.

Part of #2235.

### PR DESCRIPTION
**Pull request by JSONbored**
## Summary

Part of the mobile design-system sequence (#2235) — the first Phase-1 primitive. Replaces Material button styling with a custom `SureButton` mirroring the web `DS::Button`. **Independent** of the in-review foundation PRs (uses only tokens already on `main`).

Closes #2357.

## What changed

- **New `mobile/lib/widgets/sure_button.dart`** — `SureButton` mirroring `DS::Button` / `DS::Buttonish`:
  - variants `primary` / `secondary` / `destructive` / `outline` / `ghost`;
  - sizes `sm` / `md` / `lg` (≈ 28 / 36 / 48 high), `fullWidth`, optional `leading` widget, `loading` (adaptive spinner), disabled;
  - colors resolve from `SureColors.of(context).palette` (`buttonPrimary`/`Hover`, `buttonDestructive`/`Hover`, `surfaceInset`/`Hover`, `surfaceHover`, `borderSecondary`, `textInverse`, `textPrimary`); shape from `SureTokens.radiusMd`/`radiusLg`;
  - flat custom press feedback (swaps to the `*Hover` token, no Material ripple); a11y `Semantics(button, enabled, label)`.
- **Migrated the 3 main-view login buttons** (`mobile/lib/screens/login_screen.dart`): Sign In → primary (lg, fullWidth, loading), Sign in with Google → outline (SVG leading), API-Key Login → ghost.
- **`mobile/test/widgets/sure_button_test.dart`** — variant→token, tap, disabled, loading, and size coverage.

## Why

The web DS treats outline/ghost buttons as neutral (`text-primary`), but Material renders their foregrounds in the theme accent (blue) — visible in the before/after. `SureButton` lands one consistent, tokenized, off-Material button and is the primitive that dialogs/forms/CTAs build on.

Note on dimensions: heights/padding/font-size aren't in `sure.tokens.json` — on the web these are standard Tailwind utilities composed in `DS::Buttonish::SIZES`, not brand tokens, so `SureButton` mirrors those values directly (colors + radius remain fully tokenized).

## Screenshots

Real iOS Simulator (iPhone 17 Pro), login screen — before = Material buttons, after = `SureButton`. Note the Google/API-Key foregrounds going from Material blue to neutral DS styling.

<table>
  <thead><tr><th>Login</th><th>Before (Material)</th><th>After (SureButton)</th></tr></thead>
  <tbody>
    <tr>
      <td><strong>Light</strong></td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td><strong>Dark</strong></td>
      <td></td>
      <td></td>
    </tr>
  </tbody>
</table>

## Validation

- `cd mobile && flutter analyze --no-fatal-infos` — clean (pre-existing info-level issues only)
- `cd mobile && flutter test` — 101 passed (incl. the new `sure_button_test`)
- `cd mobile && flutter build ios --simulator --debug` + `flutter build apk --debug` — build
- CodeRabbit local review (applied: a11y "loading" label + dropped a redundant `isLoading ? null`) + adversarial review (security + correctness): PASS — no security surface, independent of unmerged PRs, login migration faithful

## Notes

- The API-key modal dialog buttons are intentionally left for a follow-up; the broader button migration (settings/transactions/chat) follows as its own slices.
- Part of #2235.

## Release Notes

* **New Features**
  * Added a reusable `SureButton` component with variants (primary/secondary/destructive/outline/ghost), sizes, optional leading content, and built-in loading/disabled behavior.
  * Updated the login screen to use `SureButton` for “Sign In”, “Sign in with Google”, and “API-key login”, with consistent loading states and actions disabled during authentication.

* **Tests**
  * Added comprehensive widget tests for `SureButton`, including variant styling, interaction behavior (disabled/loading/pressed), keyboard activation, and size/full-width layout.

* **Style**
  * Improved button press and loading feedback for authentication actions to match the new design system.

### CODE REVIEW

File: mobile/lib/widgets/sure_button.dart [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    final leading = widget.loading
+        ? SizedBox(
+            width: metrics.fontSize,
+            height: metrics.fontSize,
+            child: CircularProgressIndicator.adaptive(
+              strokeWidth: 2,
+              valueColor: AlwaysStoppedAnimation<Color>(style.foreground),
+            ),
+          )
+        : widget.leading;
```
[the remark this answers is not in the document]
Comment (JSONbored) [reply mobile/lib/widgets/sure_button.dart:110]: Good catch on the iOS tint. Heads-up though: `color:` isn't a parameter on `CircularProgressIndicator.adaptive` in the pinned Flutter **3.32.4** — that diff fails to compile (`.adaptive` only accepts `valueColor` there). So I implemented the same intent version-safely: a platform branch using `CupertinoActivityIndicator(color: …)` on iOS/macOS (which honors the tint) and `CircularProgressIndicator(valueColor: …)` on Android. The spinner now matches the button foreground on both. Fixed in eaf1242c3.

File: mobile/lib/screens/login_screen.dart [hunk 2]
```diff
@@ -513,10 +506,11 @@ class _LoginScreenState extends State<LoginScreen> {
...
 
                     // API Key Login Button
-                    TextButton.icon(
+                    SureButton(
+                      label: 'API-Key Login',
+                      variant: SureButtonVariant.ghost,
                       onPressed: _showApiKeyDialog,
-                      icon: const Icon(Icons.vpn_key_outlined, size: 18),
-                      label: const Text('API-Key Login'),
+                      leading: const Icon(Icons.vpn_key_outlined, size: 18),
```
[the remark this answers is not in the document]
Comment (JSONbored) [reply mobile/lib/screens/login_screen.dart]: Skipping this one for now, intentionally. The original `TextButton.icon` had no `isLoading` guard either, so the migration is faithful — and `SureButton` exposes disabling via `onPressed`/`loading`, not an `enabled` flag. This button only opens the API-key **dialog**, which manages its own loading/disabled state, so it isn't really an overlapping-login race. Gating it would be a small behavior change beyond this migration slice; happy to add the `Consumer` + `isLoading` guard as a follow-up if you'd prefer it gated.

File: mobile/lib/widgets/sure_button.dart [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+    final content = Row(
+      mainAxisSize: widget.fullWidth ? MainAxisSize.max : MainAxisSize.min,
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        if (leading != null) ...[leading, const SizedBox(width: 8)],
+        Flexible(child: label),
+      ],
```
[the remark this answers is not in the document]
Comment (JSONbored) [reply mobile/lib/widgets/sure_button.dart:121]: Fixed in eaf1242c3 — but kept it **conditional** rather than removing outright: `Flexible` now wraps the label only when `fullWidth` (a bounded Row), and inline (`min`-size) buttons pass the self-sizing label directly. That avoids the unbounded-constraints assertion you flagged, while still letting full-width buttons ellipsize long labels.

File: mobile/lib/screens/login_screen.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (JSONbored) [reply mobile/lib/screens/login_screen.dart]: Done in d11e03b36 — gated as suggested: wrapped the button in `Consumer<AuthProvider>` with `onPressed: authProvider.isLoading ? null : _showApiKeyDialog`, so it's disabled while an email/SSO login is in flight. Thanks!

**Review by jjmata — COMMENTED** [review]
Thorough implementation and test coverage. The mid-press/disabled edge case (`didUpdateWidget` resetting `_pressed`) and the `Flexible`-in-unbounded-Row regression are both tested, which is exactly the kind of subtle state you need coverage for in a custom button.

A few items to track as follow-ups:

**Focus ring color**: `SureButton` uses `palette.borderPrimary` for the keyboard/switch-control focus ring. PR #2349 (currently open) adds `palette.focusRing` specifically for this purpose. Once #2349 lands, `SureButton`'s focus ring should be updated to `palette.focusRing` to stay in lockstep with the design token.

**`Icons.vpn_key_outlined` on the API key button**: The API-Key Login button passes a Material `Icon` as its `leading`. Once `SureIcons` includes a key icon (or similar), this should migrate to `SureIcon` to be consistent with the rest of the design system. Not blocking, just a known loose end.

**`CupertinoActivityIndicator` on iOS**: The loading spinner correctly uses `CupertinoActivityIndicator` on Apple platforms — good native feel. The `radius: metrics.fontSize / 2` sizing will give an indicator diameter equal to the text size, which may look slightly larger than expected in some contexts (e.g. a `sm` button). Worth a quick visual pass on device.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by jjmata — APPROVED** [review]
There's a couple of follow-ups in the last Claude Code PR comment, but this is GTG!
